OSDOCS#11414: IPI improvements - Part 2 (#79537)
openshift/openshift-docs: modules/installation-azure-trusted-launch.adoc

You can enable two trusted launch features when installing your cluster on Azure: link:https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#secure-boot[secure boot] and link:https://learn.microsoft.com/en-us/windows/security/hardware-security/tpm/trusted-platform-module-overview[virtualized Trusted Platform Modules].

> For more informations about the sizes of virtual machines that support the trusted launch features, see link:https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#virtual-machines-sizes[virtual machine sizes].

Reviewer: ```suggestion
For more informations about the sizes of virtual machines that support trusted launch features, see link:https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#virtual-machines-sizes[Virtual machine sizes].
```

.Procedure

* Edit the `install-config.yaml` file before deploying your cluster:

> ** To enable trusted launch only on control plane, add the following stanza:

Reviewer: ```suggestion
** Enable trusted launch only on the control plane by adding the following stanza:
```

+
[source,yaml]
----
controlPlane:
  platform:
    azure:
      settings:
        securityType: TrustedLaunch
        trustedLaunch:
          uefiSettings:
            secureBoot: Enabled
            virtualizedTrustedPlatformModule: Enabled
----

> ** To enable trusted launch only on compute node, add the following stanza:

Reviewer: ```suggestion
** Enable trusted launch only on compute node by adding the following stanza:
```

+
[source,yaml]
----
compute:
  platform:
    azure:
      settings:
        securityType: TrustedLaunch
        trustedLaunch:
          uefiSettings:
            secureBoot: Enabled
            virtualizedTrustedPlatformModule: Enabled
----

> **  To enable trusted launch on all nodes, add the following stanza:

Reviewer: ```suggestion
**  Enable trusted launch on all nodes by adding the following stanza:
```

+
[source,yaml]
----
platform:
  azure:
    settings:

>       securityType: TrustedLaunch <1>

Reviewer 2: Here <1>/<2>/<3> seems no explanation.
The explanations are deleted by accident?

Reviewer 2: one more question: any reason to only adjust section "Enabling trusted launch for Azure VMs"? Section "Enabling confidential VMs" should have similar structure.

Author: @jinyunma I have removed  <1>/<2>/<3> . Earlier the explanations were there because we were clubbing everything into a single example. Now that it's split into 3, I think we wouldn't need that anymore.

Author: and I agree about modifying "Enabling confidential VMs". I was planning to create a separate PR. In case it's easier for you to review in this PR, I can do that
